api/end_device.proto

```diff
@@ -116,6 +118,11 @@ message MACParameters {
   // Whether downlink dwell time is set (400ms).
   // If this field is not set, then the value is either unknown or irrelevant(Network Server cannot modify it).
   google.protobuf.BoolValue downlink_dwell_time = 21;
+
+  // ADR: number of messages to wait before setting ADRAckReq.
+  ADRAckLimitExponentValue adr_ack_limit_exponent = 22 [(gogoproto.customname) = "ADRAckLimit"];
```

Comment:
Naming `adr_ack_limit_exponent` `ADRAckLimit` in Go is a bit confusing. Why can't we just call this `ADRAckLimitExponent`, and leave the old one the way it is (instead of renaming that to `DEPRECATED_ADRAckLimit`). Some "deprecated, use exponent instead" comments on those fields should be sufficient.

Reply:
Because only the type of the field changed, it still represents the exact same this - the ADRAckLimit.
Also note that we have e.g. `mac_settings.desired_adr_ack_delay`, which is exponent type as well. The user does not need to use `mac_settings.desired_adr_ack_delay_exponent`, that's just confusing, however, by looking the docs the user will see that the ADRAckLimit is set *via* exponent.
Also note that we have multiple other fields, which use enum types instead of plain uints and we don't add any extra suffixes to indicate that.
The only reason the *_exponent* suffix was added to these 2 fields is to not break existing clients using JSON.

Reply:
1. That is not true. The field now represents the exponent of the limit, not the limit itself.
2. This is something we should take as a lesson for future API design. If something is an index or exponent instead of the exact value, we should use a suffix. We can't change that now, so we'll have to move forward with this inconsistency.
3. There is absolutely no reason to make the generated name in Go different than the proto name.